stl/inc/format

```diff
@@ -637,17 +730,36 @@ _NODISCARD constexpr const _CharT* _Parse_replacement_field(
     return _Begin + 1;
 }
 
+template <class _CharT>
+const _CharT* _Scan_format_string(const _CharT* _First, const _CharT* _Last, const _CharT _Val, const _Cvtvec& _Cvt) {
+    // Returns the first occurrence of _Val as an encoded character in [_First, _Last).
+    if (_Cvt._Mbcurmax == 1 || _Cvt._Mbcurmax == 4) {
+        return _Find_unchecked(_First, _Last, _Val);
+    }
+
+    while (_First != _Last && *_First != _Val) {
+        const int _Units = _Code_units_in_next_character(_First, _Last, _Cvt);
+        if (_Units < 0) {
+            throw format_error("Invalid encoded character in format string.");

```

Comment:
I avoided throwing in `_Code_units_in_next_character` with the belief that the caller could provide a more detailed message ("malformed format string" or "malformed padding character") but I think it would be an improvement for now to throw directly, avoiding error-returns, until such time as we decide we want said more-specific messages.

Reply:
This would revert the conditional operators though, which made `_C_u_i_n_c` so much more concise. I'll wait for other opinions.

Reply:
well, if we return the error code we should at least make cuinc nodiscard

Reply:
So `[[nodiscarded]]`.